crates/js-component-bindgen/src/ts_bindgen.rs

```diff
@@ -217,36 +222,72 @@ pub fn ts_bindgen(
-                * returns a map of all the exports of the component.
-                *
-                * This function is intended to be similar to the
-                * `WebAssembly.instantiate` function. The second `imports`
-                * argument is the \"import object\" for wasm, except here it
-                * uses component-model-layer types instead of core wasm
-                * integers/numbers/etc.
-                *
-                * The first argument to this function, `compileCore`, is
-                * used to compile core wasm modules within the component.
-                * Components are composed of core wasm modules and this callback
-                * will be invoked per core wasm module. The caller of this
-                * function is responsible for reading the core wasm module
-                * identified by `path` and returning its compiled
-                * WebAssembly.Module object. This would use `compileStreaming`
-                * on the web, for example.
-                */
-                export function instantiate(
-                    compileCore: (path: string, imports: Record<string, any>) => Promise<WebAssembly.Module>,
-                    imports: ImportObject,
-                    instantiateCore?: (module: WebAssembly.Module, imports: Record<string, any>) => Promise<WebAssembly.Instance>
-                ): Promise<{camel}>;
-            ",
-        );
+    match opts.instantiation {
+        Some(InstantiationMode::Async) => {
+            uwriteln!(
+                bindgen.src,
+                "
+                    /**
+                    * Instantiates this component with the provided imports and
+                    * returns a map of all the exports of the component.
+                    *
+                    * This function is intended to be similar to the
+                    * `WebAssembly.instantiate` function. The second `imports`
+                    * argument is the \"import object\" for wasm, except here it
+                    * uses component-model-layer types instead of core wasm
+                    * integers/numbers/etc.
+                    *
+                    * The first argument to this function, `compileCore`, is
+                    * used to compile core wasm modules within the component.
+                    * Components are composed of core wasm modules and this callback
+                    * will be invoked per core wasm module. The caller of this
+                    * function is responsible for reading the core wasm module
+                    * identified by `path` and returning its compiled
+                    * `WebAssembly.Module` object. This would use `compileStreaming`
+                    * on the web, for example.
+                    */
+                    export function instantiate(
+                        compileCore: (path: string, imports: Record<string, any>) => Promise<WebAssembly.Module>,
```

Comment:
It might then make sense to do the same renaming here too.

Reply:
Done.